tests/framework/microvm.py

```diff
@@ -552,6 +589,43 @@ def start(self):
         response = self.actions.put(action_type='InstanceStart')
         assert self._api_session.is_status_no_content(response.status_code)
 
+    def pause_to_snapshot(self,
+                          mem_file_path=None,
+                          snapshot_path=None,
+                          diff=False):
+        """Pauses the microVM, and creates snapshot.
+
+        This function validates that the microVM pauses successfully and
+        returns the name of the snapshot files.
+        """
+        response = self.vm.patch(state='Paused')
+        assert self.api_session.is_status_no_content(response.status_code)
+
+        response = self.snapshot_create.put(mem_file_path=mem_file_path,
+                                            snapshot_path=snapshot_path,
+                                            diff=diff)
+        assert self.api_session.is_status_no_content(response.status_code)
+        return mem_file_path, snapshot_path
+
+    def resume_from_snapshot(self, mem_file_path, snapshot_path):
```

Comment:
Maybe validate that `mem_file_path` and `snapshot_path` are not `None`?

Reply:
This is useful for negative tests.

Reply:
However there asserts do not allow for this and will fail if one of these paths is None.